packages/squiggle-lang/src/rescript/Distributions/PointSetDist/PointSetDist.res

```diff
@@ -195,25 +198,16 @@ module T = Dist({
     | Discrete(m) => Discrete.T.variance(m)
     | Continuous(m) => Continuous.T.variance(m)
     }
-
-  let klDivergence = (prediction: t, answer: t) =>
-    switch (prediction, answer) {
-    | (Continuous(t1), Continuous(t2)) => Continuous.T.klDivergence(t1, t2)
-    | (Discrete(t1), Discrete(t2)) => Discrete.T.klDivergence(t1, t2)
-    | (m1, m2) => Mixed.T.klDivergence(m1->toMixed, m2->toMixed)
-    }
-
-  let logScoreWithPointResolution = (~prediction: t, ~answer: float, ~prior: option<t>) => {
-    switch (prior, prediction) {
-    | (Some(Continuous(t1)), Continuous(t2)) =>
-      Continuous.T.logScoreWithPointResolution(~prediction=t2, ~answer, ~prior=t1->Some)
-    | (None, Continuous(t2)) =>
-      Continuous.T.logScoreWithPointResolution(~prediction=t2, ~answer, ~prior=None)
-    | _ => Error(Operation.NotYetImplemented)
-    }
-  }
 })
 
+let logScore = (args: PointSetDist_Scoring.scoreArgs): result<float, Operation.Error.t> =>
+  PointSetDist_Scoring.logScore(
+    args,
+    ~combineFn=combinePointwise,
+    ~integrateFn=T.Integral.sum,
+    ~toMixedFn=toMixed,
```

Comment:
It's fairly awkward that this function is also used for scalars now. I imagine we could just remove it from here, because of that. Hypothetically we could create a version just for PointSetDist versions, but that seems like too much work.

Reply:
I suppose it wouldn't mean nonexhaustively pattern matching on each half, or splitting up `scoreArgs` into two different types, because you could have half the cases result in `Error` for the function that acts on `PointSetDist` and the other half of the cases result in `Error` for the function that acts on scalars.